source/uri-options/tests/README.rst

```diff
@@ -0,0 +1,55 @@
+=======================
+URI Options Tests
+=======================
+
+The YAML and JSON files in this directory tree are platform-independent tests
+that drivers can use to prove their conformance to the URI Options spec.
+
+These tests use the same format as the Connection String spec tests.
+
+Version
+-------
+
+Files in the "specifications" repository have no version scheme. They are not
+tied to a MongoDB server version.
+
+Format
+------
+
+Each YAML file contains an object with a single ``tests`` key. This key is an
+array of test case objects, each of which have the following keys:
+
+- ``description``: A string describing the test.
+- ``uri``: A string containing the URI to be parsed.
+- ``valid``: A boolean indicating if the URI should be considered valid. 
+  This will always be true, as the Connection String spec tests the validity of the structure, but 
+  it's still included to make it easier to reuse the connection string spec test runners that 
+  drivers already have.
+- ``warning``: A boolean indicating whether URI parsing should emit a warning.
+- ``hosts``: An array of host objects, each of which have the following keys:
+  - ``type``: Included for compatibility with the Connection String spec tests. This will always be ``~``.
+  - ``host``: Included for compatibility with the Connection String spec tests. This will always be ``~``.
+  - ``port``: Included for compatibility with the Connection String spec tests. This will always be ``~``.
+- ``auth``: An object containing the following keys:
+  - ``username``: Included for compatibility with the Connection String spec tests. This will always be ``~``.
+  - ``password``: Included for compatibility with the Connection String spec tests. This will always be ``~``.
+  - ``db``: Included for compatibility with the Connection String spec tests. This will always be ``~``.
+- ``options``: An object containing key/value pairs for each parsed query string
+  option.
+
+If a test case includes a null value for one of these keys (e.g. ``auth: ~``,
```

Comment:
@saghm can comment as to why we have the keys within `hosts`/`auth` present when all of them are `null`? In the connection string spec, the top-level key is itself set to `null` when it doesn't contain any information. To re-use the test-runner we have in python, I would need to do the same here. That said, I am not averse to modifying the runner to be able to handle this pattern provided we have a good rationale for stroing `null`-valued keys.

Reply:
Interesting, I must have missed the specs that had the hosts field as null; I'll update this to do that

Reply:
Just pushed the changes